AWSAuthSDK/AWSAuthSDKTestApp/UserDetailsViewController.swift

```diff
@@ -58,9 +63,19 @@ class UserDetailsViewController: UIViewController {
             self.credentialExpirationLabel.text = "NA"
             self.attribute1Label.text = "NA"
             self.attribute2Label.text = "NA"
+            self.attribute3Label.text = "NA"
+            self.customeAttribute1Label.text = "NA"
+            self.customeAttribute2Label.text = "NA"
         }
     }
-    
+
+    func invalidateSessionOrNot() {
```

Comment:
This is bit confusing, can we instead add a button in the UI called "invalidateRefreshToken"? And the test will tap on it to invalidate refresh token?

Reply:
I have thought about that way. The problem or the difficult part is `UserDetailsViewController` does not have context on which text is invoking it. The `fetchUserAttribute()` is called whenever the `viewDidLoad()` is called. So that my workaround here is for test `testHostedUIGetAttributesWhenSessionExpired()`, create a user prefixed with `sessionExpired`, and once `UserDetailsViewController` is presented, it checks the prefix of username and then know whether should call `invalidateSession` or not. I can put some comments to explain the purpose of this method.